src/libs/kata-types/src/config/mod.rs

```diff
@@ -94,21 +94,15 @@ impl TomlConfig {
     /// If `config_file` is valid, it will used, otherwise a built-in default path list will be
     /// scanned.
     pub fn load_from_file<P: AsRef<Path>>(config_file: P) -> Result<(TomlConfig, PathBuf)> {
-        let file_path = if !config_file.as_ref().as_os_str().is_empty() {
-            fs::canonicalize(config_file)?
-        } else {
-            Self::get_default_config_file()?
-        };
-
-        info!(
-            sl!(),
-            "load configuration from: {}",
-            file_path.to_string_lossy()
-        );
-        let content = fs::read_to_string(&file_path)?;
-        let config = Self::load(&content)?;
+        let mut result = Self::load_raw_from_file(config_file);
+        if let Ok((ref mut config, _)) = result {
+            Hypervisor::adjust_config(config)?;
```

Comment:
I believe it should be, fundamentally.  However, `load()` is used to initialise `KATA_DEFAULT_CONFIG` and `KATA_ACTIVE_CONFIG` and I wasn't sure what the ramifications would be if I messed with `load()`'s implementation.  

Also, if `load()` were to handle drop-ins as well its interface would need to change - it would have to take a _path_ to the base config file, not just its _contents_ as it's impossible to figure out where to look for drop-ins given just the base config file contents.

That said I'm very open to suggestions how best to integrate drop-in handling into existing config loading code.  I'm not at all sure the path I've taken is the best one.

Reply:
The implementation looks good so far